solana/solana-ibc/programs/solana-ibc/src/transfer/impls.rs

```diff
@@ -388,17 +436,26 @@ impl IbcStorage<'_, '_> {
             .filter(|escrow_account| escrow.eq(escrow_account.key))
             .ok_or(TokenTransferError::ParseAccountFailure)?;
 
-        accounts
-            .token_account
-            .as_ref()
-            .filter(|token_account| account.0.eq(token_account.key))
-            .ok_or(TokenTransferError::ParseAccountFailure)?;
-
+        // We only need to check for sender/receiver since the token account
+        // is always derived from the token mint so if sender/receiver are right,
+        // the token account would be right as well.
         let ok = match op {
```

Comment:
I’m not too happy with this match here.  On one side we have `ok_or`
with question mark operator.  On the other the `ok` variable is still
used.  Then again I don’t know if there’s some cleaner way to do it.
Perhaps something like:

```
        match op {
            EscrowOp::Escrow => {
                accounts
                    .sender
                    .as_deref()
                    .filter(|sender| sender.is_signer)
            },
            EscrowOp::Unescrow => {
                accounts
                    .receiver
                    .as_deref()
                    .filter(|_| accounts.mint_authority.is_some())
            },
        }
            .filter(|acc| account.0 == *acc.key)
            .map(|_| ())
            .ok_or(TokenTransferError::ParseAccountFailure)
```

Reply:
i think above is good enough